homeassistant/components/scrape/sensor.py

```diff
@@ -105,7 +133,7 @@ async def async_setup_platform(
     await rest.async_update()
 
     if rest.data is None:
-        raise PlatformNotReady
+        raise ConfigEntryNotReady
```

Comment:
I don't think this will work.  The config entry will already be setup at this point.

It needs to be raised in `__init__.py` `async_setup_entry`

Reply:
We should have a test that the config entry gets to a SETUP_RETRY state to be sure this works

Reply:
It doesn't work as it comes back as loaded.
So should move this to __init__ or it's quite pointless to have.